--- a/drivers/net/ethernet/mediatek/mtk_eth_dbg.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_dbg.c
@@ -2866,7 +2866,7 @@ int mtk_eth_debugfs_init(struct mtk_eth
 		}
 		eth->debugfs->qdma_sched[i].eth = eth;
 		eth->debugfs->qdma_sched[i].id = i;
-		debugfs_create_file(name, 0444, eth->debugfs->root,
+		debugfs_create_file(name, 0644, eth->debugfs->root,
 				    (void *)&eth->debugfs->qdma_sched[i],
 				    &mtk_eth_debugfs_qdma_sched_fops);
 	}
@@ -2879,7 +2879,7 @@ int mtk_eth_debugfs_init(struct mtk_eth
 		}
 		eth->debugfs->qdma_queue[i].eth = eth;
 		eth->debugfs->qdma_queue[i].id = i;
-		debugfs_create_file(name, 0444, eth->debugfs->root,
+		debugfs_create_file(name, 0644, eth->debugfs->root,
 				    (void *)&eth->debugfs->qdma_queue[i],
 				    &mtk_eth_debugfs_qdma_queue_fops);
 	}
